nova/virt/vmwareapi/driver.py

```diff
@@ -63,6 +65,12 @@
 MAX_CONSOLE_BYTES = 100 * units.Ki
 
 
+def _required_attribute(obj, attr):
```

Comment:
This is used only in `check_can_live_migrate_source` so maybe is better to move it down locally (especially that it throws a MigrationPreCheckError.

Reply:
Personally, I prefer free functions over nested ones, as they have less context, but I have changed it nonetheless. I see your point too.